olive/engine/footprint.py

```diff
@@ -422,3 +429,46 @@ def _plot_pareto_frontier(self, ranks=None, save_path=None, is_show=True, save_f
 
         if is_show:
             fig.show()
+
+
+def get_best_candidate_node(
```

Comment:
how does this work wrt to no-search model with multiple workflows? In these cases, we probably won't have any metrics for the output models. Does this just select the first output model? If so, is using multiple flows without evaluator not supported since the other flows are ignored? Maybe it's fine since Devang did not want pass_flows to be exposed to the user.

Reply:
For no search, there is one condition before calling this func in Engine: `if outputs and self.passes and not next(iter(outputs.values())).check_empty_nodes()`. No search has no node and the output is `MetricResult`. Engine will directly return the outputs.

Reply:
I don't think that's true? In line 334 you can see that no search also returns a output with the nodes for the final models for each workflow. so, this won't be True. 
Otherwise, it would mean the output model is never saved in no search mode.

Reply:
Yeah i just double checked and you are right. I don't remember in which case that the result is MetricResult but from what i tested, there will be only one node in pf footprints for no search case and that one will be the output one.

Reply:
with multiple flows, I think there will be multiple nodes? in that case, if there is no evaluator, does it choose the first output model? If so, is this the behavior we want?

Reply:
could you verify the output structure of the combination of the all the following scenarios?
- search/no-search(with/out accelerator)
- one/multiple accelerators
- one/multiple pass flows

Reply:
We won't cover the case for no search + multi pass flow as pass flow will be deprecated with no search soon. So the output structures are:

- search/no-search(with/out evaluator)
 - search will copy the best node model files to output folder.
 - no search will copy the last node files to the output folder (with/out evaluator)
- one/multiple accelerators
 - will copy the best node models to the output folder, with device-ep folder with run history json file.
- one/multiple pass flows
 - will copy the best node models to the output folder, with pass_flow folder (if has) for metric json file.

Reply:
Most of them are correct. I removed multi pass flow case for no search, and added final output model path description.

Reply:
thanks! sorry for taking long to review. wanted to make sure all scenarios were covered since some the cli and demos depend on the output structure.